CLI/support/sparkle_job_help.py

```diff
@@ -125,17 +128,84 @@ def get_running_jobs() -> list[SlurmRun]:
             if run.status == Status.WAITING or run.status == Status.RUNNING]
 
 
+def get_dependencies(jobs: list[SlurmRun]) -> list[SlurmRun]:
+    """Adds Dependencies. Should be removed once RunRunner provides this feature."""
+    for job in jobs:
+        with job.json_filepath.open() as file:
+            data = json.load(file)
+        if len(data["dependencies"]) > 0:
+            job_ids = [dep["run_id"] for dep in data["dependencies"]]
+        else:
+            job_ids = []
+        job.dependencies = job_ids
+    return jobs
+
+
+def clear_console_lines(lines: int) -> None:
+    """Clears the last lines of the console."""
+    # \033 is the escape character (ESC) in ASCII
+    # [{lines}A is the escape sequence that moves the cursor up.
+    print(f"\033[{lines}A", end="")
+    # [J is the exape sequence that clears the console from the cursor down
+    print("\033[J", end="")
+
+
 def wait_for_all_jobs() -> None:
     """Wait for all active jobs to finish executing."""
-    running_jobs = get_running_jobs()
-    prev_jobs = len(running_jobs) + 1
-    while len(running_jobs) > 0:
-        if len(running_jobs) < prev_jobs:
-            print(f"Waiting for {len(running_jobs)} jobs...", flush=True)
-        time.sleep(10.0)
-        prev_jobs = len(running_jobs)
-        running_jobs = [run for run in running_jobs
-                        if run.status == Status.WAITING or run.status == Status.RUNNING]
+    jobs = get_running_jobs()
+    verbosity_setting = gv.settings.get_general_verbosity()
+    running_jobs = [run for run in jobs
```

Comment:
Why is this done when we're already calling a function get_running_jobs?

Reply:
We seperate between jobs and running_jobs to know when we're done (when running_jobs == 0)
However you're right that there is no need to check again in line 157 for running jobs as they all should be running or pending.
I'll change it to running_jobs = jobs

Reply:
Can we change it into running_jobs = get_running_jobs()? And skip the two variables into 1?